### PR TITLE
Ntfy provider

### LINKED ISSUE #4185: 🚀 Feature: Support ntfy
### 🔖 Feature description

Support ntfy as Push dest

### 🎤 Why is this feature needed ?

ntfy is a free and widely-sued

### ✌️ How do you aim to achieve this?

With mobile to receive notificaiton

### 🔄️ Additional Information

_No response_

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Contributing Guidelines](https://github.com/novuhq/novu/blob/main/CONTRIBUTING.md)

### Are you willing to submit PR?

None

### PR DESCRIPTION
**Pull request by sandsower**
### What change does this PR introduce?

Adds ntfy support as a push provider

### Why was this change needed?

Closes #4185

### Other information (Screenshots)

Continues work started by @ayushdiitr way back in October, this fixes the configuration and finishes up the setup

Screenshot 2024-03-29 at 12-24-55 Integrations Novu Manage Platform

### CODE REVIEW

File: apps/web/Dockerfile
Comment (BiswaViraj) [inline apps/web/Dockerfile:1]: ❓ **question:** Why these changes are needed?‏

**Review by scopsy — CHANGES_REQUESTED** [review]

File: packages/application-generic/src/factories/push/handlers/ntfy.handler.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+export class NtfyHandler extends BasePushHandler {
+  constructor() {
+    super('ntfy', ChannelTypeEnum.PUSH);
+  }
+
+  buildProvider(credentials: ICredentials) {
+    if (!credentials.baseUrl || !credentials.topic) {
+      throw Error('Config is not valid for ntfy');
+    }
```
Comment (BiswaViraj) [inline packages/application-generic/src/factories/push/handlers/ntfy.handler.ts:13]:
```suggestion
    if (!credentials.topic) {
      throw Error('Topic is required for ntfy');
    }
```


⚠ **issue:** As `baseUrl` is optional, it shouldn't throw error when `baseUrl` is empty‏

**Review by BiswaViraj — CHANGES_REQUESTED** [review]

File: apps/web/Dockerfile
Comment (sandsower) [reply apps/web/Dockerfile:1]: Mistake on my part for testing, reverting it

File: docker/local/deployment/docker-compose.yml [hunk 2]
```diff
@@ -181,5 +185,25 @@ services:
...
+      - /etc/ntfy:/etc/ntfy
+    ports:
+      - 3008:80
+    healthcheck: # optional: remember to adapt the host:port to your environment
+        test: ["CMD-SHELL", "wget -q --tries=1 http://localhost:80/v1/health -O - | grep -Eo '\"healthy\"\\s*:\\s*true' || exit 1"]
+        interval: 60s
+        timeout: 10s
+        retries: 3
+        start_period: 40s
+    restart: unless-stopped
```
[the remark this answers is not in the document]
Comment (sandsower) [reply docker/local/deployment/docker-compose.yml]: Yeah this was some code leftover from testing, I've reverted it
